src/components/WalletBalance/index.tsx

```diff
@@ -1,51 +1,54 @@
-import React, {Fragment, useEffect, useState} from 'react'
+import React from 'react'
 import { ChainId, TokenAmount } from '@sushiswap/sdk'
-import { AdditionalChainId } from '../../constants'
-import { CustomLightSpinner } from 'theme'
-import Circle from 'assets/images/blue-loader.svg'
+import { Fragment, useEffect, useState } from 'react'
 import { LoadingPlaceHolder } from 'theme/components'
+import { AdditionalChainId } from '../../constants'
 
 export interface Balances {
-  G$: TokenAmount | undefined,
-  GDX: TokenAmount | undefined,
-  GOOD: TokenAmount | undefined
+    G$: TokenAmount | undefined
+    GDX: TokenAmount | undefined
+    GOOD: TokenAmount | undefined
+}
+
+export interface BalancesV2 {
+    G$: any
+    GOOD: any
 }
 
 export type WalletBalanceProps = {
-  balances: Balances,
-  chainId: ChainId,
+    balances: BalancesV2
+    chainId: ChainId
 }
 
 export default function WalletBalance(props: WalletBalanceProps): JSX.Element {
-  const { balances, chainId } = props
-  const [balance, setBalance] = useState<JSX.Element[]>()
+    const { balances, chainId } = props
+    const [balance, setBalance] = useState<JSX.Element[]>()
 
-  useEffect(() => {
-    const newBalance = Object.entries(balances).map((balance) => {
-      const token = balance[0]
-      const amount = balance[1]
-      if ((chainId as any) === AdditionalChainId.FUSE && token === 'GDX') return <div key={token}></div>
-      const frag = (
-        <Fragment key={token}>
-          <span className="flex">
-            {token} - {
-              !amount ?
-                <LoadingPlaceHolder /> :
-                token === 'GOOD' ?
-                  amount.toSignificant(6, {groupSeperator: ','}) :
-                  amount.toExact({ groupSeperator: ','})
-            }
-          </span>
-        </Fragment>
-      )
-      return frag
-    })
-    setBalance(newBalance)
-  }, [chainId, balances])
+    useEffect(() => {
+        if (balances) {
+            const newBalance = Object.entries(balances).map((balance) => {
+                const token = balance && balance[0]
+                const amount = balance && balance[1]?.amount
+                if (Object.values(AdditionalChainId).includes(chainId as any) && token === 'GDX')
+                    return <div key={token}></div>
+                const frag = (
+                    <Fragment key={token}>
+                        <span className="flex">
+                            {!amount ? (
+                                <LoadingPlaceHolder />
+                            ) : (
+                                amount.format({ suffix: '', prefix: amount.currency.ticker + ' - ' })
```

Comment:
```suggestion
                                amount?.format({ suffix: '', prefix: amount?.currency.ticker + ' - ' })
```

also could be moved to useMemo